uaclient/cli.py

```diff
@@ -197,155 +272,289 @@ def _perform_enable(entitlement_name: str, cfg: config.UAConfig, *,
-            contract_name=contract_name))
+            contract_name=contract_name
+        )
+    )
 
     action_status(args=None, cfg=cfg)
     return 0
 
 
 def get_parser():
+    service_line_tmpl = " - {name}: {description}{url}"
+    description_lines = [__doc__]
+    sorted_classes = sorted(entitlements.ENTITLEMENT_CLASS_BY_NAME.items())
+    for name, ent_cls in sorted_classes:
+        if ent_cls.help_doc_url:
+            url = " ({})".format(ent_cls.help_doc_url)
+        else:
+            url = ""
+        service_line = service_line_tmpl.format(
+            name=name, description=ent_cls.description, url=url
+        )
+        if len(service_line) <= 80:
+            description_lines.append(service_line)
+        else:
+            wrapped_words = []
+            line = service_line
+            while len(line) > 80:
+                [line, wrapped_word] = line.rsplit(" ", 1)
+                wrapped_words.insert(0, wrapped_word)
+            description_lines.extend([line, "   " + " ".join(wrapped_words)])
+
     parser = argparse.ArgumentParser(
-        prog=NAME, formatter_class=argparse.RawDescriptionHelpFormatter,
-        description=__doc__,
-        usage=USAGE_TMPL.format(name=NAME, command='[command]'),
-        epilog=EPILOG_TMPL.format(name=NAME, command='[command]'))
+        prog=NAME,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description="\n".join(description_lines),
+        usage=USAGE_TMPL.format(name=NAME, command="[command]"),
+        epilog=EPILOG_TMPL.format(name=NAME, command="[command]"),
+    )
     parser.add_argument(
-        '--debug', action='store_true',
-        help='Show all debug log messages to console')
-    parser._optionals.title = 'Flags'
+        "--debug",
+        action="store_true",
+        help="show all debug log messages to console",
+    )
+    parser._optionals.title = "Flags"
     subparsers = parser.add_subparsers(
-        title='Available Commands', dest='command', metavar='')
+        title="Available Commands", dest="command", metavar=""
+    )
     subparsers.required = True
     parser_status = subparsers.add_parser(
-        'status', help='current status of all ubuntu advantage services')
+        "status", help="current status of all Ubuntu Advantage services"
+    )
     parser_status.set_defaults(action=action_status)
     status_parser(parser_status)
     parser_attach = subparsers.add_parser(
-        'attach',
-        help='attach this machine to an ubuntu advantage subscription')
+        "attach",
+        help="attach this machine to an Ubuntu Advantage subscription",
+    )
     attach_parser(parser_attach)
     parser_attach.set_defaults(action=action_attach)
+    parser_attach_premium = subparsers.add_parser(
+        "attach-premium",
+        help=(
+            "for premium ubuntu images, attach this machine to an ubuntu"
+            " advantage subscription"
+        ),
```

Comment:
Update help to: "automatically enable Ubuntu Advantage on a premium Ubuntu image"